### PR TITLE
fix meta index name bug

### PR DESCRIPTION
**Pull request by MRossol**
fix meta being named 'gid' during collect

### CODE REVIEW

File: reV/qa_qc/summary.py [hunk 1]
```diff
@@ -207,7 +207,9 @@ def summarize_means(self, out_path=None):
         with Resource(self.h5_file, group=self._group) as f:
             meta = f.meta
             if 'gid' not in meta:
-                meta.index.name = 'gid'
+                if meta.index.name == 'gid':
+                    meta.index.name = 'gid'
```
Comment (grantbuster) [inline reV/qa_qc/summary.py:211]: seems redundant?

Comment (MRossol) [reply reV/qa_qc/summary.py:211]: whoops, good call, was trying to Not run that unless we need to

**Comment by grantbuster** [comment]
@MRossol can you also patch this while you're at it? This warn message got put in the wrong place somehow. It gets triggered whenever `analysis_years` is input to the config as a list, which is nearly always. Maybe it should just be:
```
            self._years = self.get('analysis_years', [None])
            if not isinstance(self._years, list):
                self._years = [self._years]
            if self._years[0] is None:
                warn()
```

**Comment by MRossol** [comment]
Done

**Pull request merged**